Description: Purge unused features and dependencies
 This fork has the sole scope of providing syntect's functionality to the
 broot crate without making it panick on regex mismatches. In the whole rust
 ecosystem only its `default-fancy` feature is used. Moreover, neither the
 fork nor the original crate take much care in correctly defining the feature
 dependencies. This patch disables unused features and dependencies to ease
 packaging.
Author: NoisyCoil <noisycoil@tutanota.com>
Forwarded: no
Last-Update: 2025-01-02
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -131,11 +131,6 @@
 [dependencies.once_cell]
 version = "1.8"
 
-[dependencies.onig]
-version = "6.0"
-optional = true
-default-features = false
-
 [dependencies.plist]
 version = "1.3"
 optional = true
@@ -192,7 +187,7 @@
 version = "0.1.5"
 
 [features]
-default = ["default-onig"]
+default = ["default-fancy"]
 default-fancy = [
     "parsing",
     "default-syntaxes",
@@ -204,45 +199,29 @@
     "dump-create",
     "regex-fancy",
 ]
-default-onig = [
-    "parsing",
-    "default-syntaxes",
-    "default-themes",
-    "html",
-    "plist-load",
-    "yaml-load",
-    "dump-load",
-    "dump-create",
-    "regex-onig",
-]
 default-syntaxes = [
     "parsing",
     "dump-load",
 ]
 default-themes = ["dump-load"]
 dump-create = [
-    "flate2",
-    "bincode",
+    "dep:flate2",
+    "dep:bincode",
 ]
 dump-load = [
-    "flate2",
-    "bincode",
+    "dep:flate2",
+    "dep:bincode",
 ]
 html = ["parsing"]
-metadata = [
-    "parsing",
-    "plist-load",
-]
 parsing = [
-    "regex-syntax",
-    "fnv",
+    "dep:regex-syntax",
+    "dep:fnv",
     "dump-create",
     "dump-load",
 ]
-plist-load = ["plist"]
+plist-load = ["dep:plist"]
 regex-fancy = ["fancy-regex"]
-regex-onig = ["onig"]
 yaml-load = [
-    "yaml-rust",
+    "dep:yaml-rust",
     "parsing",
 ]
